library/src/commonMain/kotlin/com/chrisjenx/yakcov/ValueValidator.kt

```diff
@@ -192,27 +194,35 @@ abstract class ValueValidator<V, R>(
      *  or if loosing focus when [validateOnFocusLost] is `true`.
      *  When you call [validate] it will start validating showing errors if present.
      */
-    @Composable
     fun Modifier.validationConfig(
         validateOnFocusLost: Boolean = false,
         shakeOnInvalid: Boolean = false,
         showErrorOnInteraction: Boolean = true,
-    ): Modifier {
+    ): Modifier = composed {
         // Set scope if shaking is enabled
         shakeOnInvalidScope = if (shakeOnInvalid) rememberCoroutineScope() else null
         // Track if we should show error on user input
         showErrorOnUserInput = showErrorOnInteraction
-        // Track focus
-        var hadFocus by mutableStateOf(false)
-        return this
-            .onFocusChanged { focusState ->
-                if (focusState.hasFocus) hadFocus = true
+        val validateOnFocusLostModifier = if (validateOnFocusLost) {
+            var hadFocus by remember { mutableStateOf(false) }
```

Comment:
ahh yeah good catch, I thought I saw something odd on reimbursements, but never tracked it down